src/wodin.ts

```diff
@@ -0,0 +1,168 @@
+import { Dust } from "./dust";
+import type { DustModel, DustModelInfo, DustModelConstructable, UserType } from "./model";
+import type { DustStateTime } from "./state-time";
+import { isEqualArray, meanArray, seq } from "./util";
+
+/**
+ * Describes the role that each series plays
+ */
+export enum DiscreteSeriesMode {
+    /** An individual stochastic trace */
+    Individual = "Individual",
+    /** The mean over several Individual trace */
+    Mean = "Mean",
+    /** A single trace for where no variation was observed across all replicates (particles) */
+    Deterministic = "Deterministic"
+}
+
+/**
+ * Interface for objects returned by {@link wodinRunDiscrete}. This
+ * represents some number of model variables, each of which may
+ */
+export interface DiscreteSeriesSet {
```

Comment:
It feels a little bit unJSONy to have all these arrays which just happen to match each other by index. Given that we're going to have to reshape these anyway for plotly, would it be a little friendlier to have something like 
```
export interface DiscreteSeriesValues {
  name: string,
  y: number[],
  mode: DiscreteSeriesMode
}

export interface DiscreteSeriesSet {
  x: number
  values: DiscreteSeriesValues[]
}
``` 

I guess that would be inconsistent with what we get from the continuous models though, and maybe there's a reason why the current structure is easier to deal with?

Reply:
I can do this, sure, no strong reason for the current format at all!